javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java

```diff
@@ -255,6 +255,12 @@ private void inferTypes(ResolvedType source, ResolvedType target, Map<ResolvedTy
             mappings.put(target.asTypeParameter(), source);
             return;
         }
+        if (source.isArray() && target.isArray()) {
+            ResolvedType sourceComponentType = source.asArrayType().getComponentType();
+            ResolvedType targetComponentType = target.asArrayType().getComponentType();
```

Comment:
The component type for `int a[][][]` is `int[][]`, and its element type is `int`. Does this code work for multidimensional arrays too?

Reply:
Good enough for me :-)